src/sql/workbench/parts/objectExplorer/browser/iconRenderer.ts

```diff
@@ -0,0 +1,304 @@
+		colorConnected?: string, colorDisconnected?: string, colorBackground?: string
+	) {
+		super(iconSizePx, topMarginPx, leftMarginPx);
+		this.setColorConnected(colorConnected);
+		this.setColorDisconnected(colorDisconnected);
+		this.setColorBackground(colorBackground);
+	}
+
+	public setColorConnected(color: string): ServerStatusBadge {
+		this.colorConnected = color || this.colorConnected;
+		return this;
+	}
+
+	public setColorDisconnected(color: string): ServerStatusBadge {
+		this.colorDisconnected = color || this.colorDisconnected;
+		return this;
+	}
+
+	public setColorBackground(color: string): ServerStatusBadge {
+		this.colorBackground = color || this.colorBackground;
+		return this;
+	}
+
+	public renderBadge(element: HTMLElement, isConnected: boolean): void {
+		let circleColor: string = isConnected ? this.colorConnected : this.colorDisconnected;
+		let backgroundColor: string = isConnected ? this.colorConnected : this.colorBackground;
+		ServerStatusBadge.renderBadgeInternal(element, circleColor, backgroundColor,
+			this.iconSizePx, this.topMarginPx, this.leftMarginPx);
+	}
+
+	public static renderBadgeInternal(element: HTMLElement, circleColor: string, backgroundColor?: string,
+		iconSizePx?: number, topMarginPx?: number, leftMarginPx?: number
+	) {
+		backgroundColor = backgroundColor || ServerStatusBadge.defaultColorBackground;
+		iconSizePx = iconSizePx || ServerStatusBadge.defaultIconSizePx;
+		topMarginPx = topMarginPx || ServerStatusBadge.defaultTopMarginPx;
+		leftMarginPx = leftMarginPx || ServerStatusBadge.defaultLeftMarginPx;
+
+		let height: string = `${iconSizePx / 16 * 0.25}rem`;
+		let width: string = height;
+		let verticalLocation: string = `${(iconSizePx / 16 * 9) + topMarginPx}px`;
+		let horizontalLocation: string = `${(iconSizePx / 16 * 14) + leftMarginPx}px`;
+		let borderWidth: string = `${iconSizePx / 16 * 0.12}rem`;
+
+		let styleIdPrefix: string = 'ServerStatusBadgeStyle_';
+		let badgeId: string = `id_${generateUuid()}`;
+		let styleId: string = `${styleIdPrefix}${badgeId}`;
+		super.removeExistingBadgeByPrefix(element, styleIdPrefix);
+
+		element.innerHTML = (element.innerHTML || '') +
+			`<style type="text/css" id="${styleId}">
+				#${badgeId}:after {
+					position: absolute;
+					height: ${height};
+					width: ${width};
+					top: ${verticalLocation};
+					left: ${horizontalLocation};
+					border: ${borderWidth} solid ${circleColor};
+					border-radius: 100%;
+					background: ${backgroundColor};
+					content:"";
+					font-size: 100%;
+					line-height: 100%;
+					color:white;
+					text-align:center;
+					vertical-align:middle;
+				}
+			</style>
+			<div style="width: 0px; height: 0px;" id="${badgeId}">
+			</div>`.replace(/\t/g, ' ').replace(/\r?\n/g, ' ').replace(/  +/g, ' ');
+	}
+
+	public static removeExistingBadge(element: HTMLElement) {
+		super.removeExistingBadgeByPrefix(element, 'ServerStatusBadgeStyle_');
+	}
+}
+
+export class NewBadge extends BaseBadge {
```

Comment:
This class isn't used anywhere - I'm guessing you had it for testing & verification. Please remove.

Reply:
fixed.